doc/StructuredTypes.md

```diff
@@ -0,0 +1,185 @@
+## Concept
+
+Snowflake offers a way to store structured types which can be:
```

Comment:
would it make sense to link relevant snowflake docs? https://docs.snowflake.com/en/sql-reference/data-types-structured

Reply:
Right, I added reference to the Snowflake docs.